src/include/btree_inline.h

```diff
@@ -1524,27 +1556,55 @@ __wt_ref_block_free(WT_SESSION_IMPL *session, WT_REF *ref)
 }
 
 /*
- * __wt_page_del_active --
- *     Return if a truncate operation is active.
+ * __wt_page_del_visible --
+ *     Return if a truncate operation is visible to the caller.
  */
 static inline bool
-__wt_page_del_active(WT_SESSION_IMPL *session, WT_REF *ref, bool visible_all)
+__wt_page_del_visible(WT_SESSION_IMPL *session, WT_PAGE_DELETED *page_del, bool visible_all)
 {
-    WT_PAGE_DELETED *page_del;
     uint8_t prepare_state;
 
-    WT_ASSERT(session, ref->state == WT_REF_LOCKED);
+    /*
+     * In general usage, a NULL WT_PAGE_DELETED is a truncate operation whose details were discarded
+     * when it became globally visible.
+     */
+    if (page_del == NULL)
+        return (true);
+
+    /* We discard page_del on transaction abort, so should never see an aborted one. */
+    WT_ASSERT(session, page_del->txnid != WT_TXN_ABORTED);
 
-    if ((page_del = ref->ft_info.del) == NULL)
-        return (false);
-    if (page_del->txnid == WT_TXN_ABORTED)
-        return (false);
     WT_ORDERED_READ(prepare_state, page_del->prepare_state);
     if (prepare_state == WT_PREPARE_INPROGRESS || prepare_state == WT_PREPARE_LOCKED)
-        return (true);
```

Comment:
Why we think prepared truncate is visible before? Is it just a bug?

Reply:
That change inverts the sense of the function. `__wt_page_del_visible` returns true if the deletion is visible; `__wt_page_del_active` returns true if the deletion _isn't_ visible. I was planning to make that change its own changeset so it could be audited carefully, because it's really easy to mess up, but we needed it in here.

It is all supposed to be the same as before given that flip. If you see that it isn't somewhere, scream loudly :-)